### PR TITLE
[DOC] Add multidata source Dev Tools integration

### LINKED ISSUE #3452: [DOC] OSD Multiple Datasource integration features for GA release
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
OSD Multiple Datasource(MD) was released in 2.4 as an experimental feature. We plan to deliver a couple of small and medium features to bring MD out of experimental. These features will keep improving fundamental use cases of MD in OSD components (index-pattern, visualization, dev tool console, etc)

Key features are 
1) distinguish index pattern from different datasources, in all related OSD functionalities. 
2) Send request against datasources from dev tool console

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
meta issue link in OSD repo: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3577

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Add new documentation on multidata source Dev Tools integration and create new TOC section Dev Tools

### Issues Resolved
#3452 

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3577

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _dashboards/discover/multi-data-sources.md [hunk 1]
```diff
@@ -132,6 +132,24 @@ The following image shows a date histogram with an interval dropdown list.
...
+Selecting multiple data sources in the Dev Tools console allows you to work with a broader range of data and gain deeper insights into your code and applications. You can follow these steps:
+
+1. To access the Dev Tools console, open OpenSearch Dashboards and select **Dev Tools** in the main menu.
+2. In the console, input the following query in the editor pane and then select the play button.
+
+    ```json
+    GET /_cat/indices
+    ```
+
+3. From the **DataSource** dropdown menu, select a data source and then query the source.
```
Comment (zhongnansu) [inline _dashboards/discover/multi-data-sources.md:146]: Note that the datasource dropdown will appear only when datasource feature is enabled in setting yml file. Like this section `# Modifying the multiple data sources settings` mentions.

**Review by zhongnansu — APPROVED** [review]
Overall lgtm. Left 1 non blocking comment. Thanks

File: _dashboards/discover/multi-data-sources.md (hunk 1 above)
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:146]: Added steps to modify the YAML configuration.

**Comment by vagimeli** [comment]
> Overall lgtm. Left 1 non blocking comment. Thanks

@zhongnansu Revised to add steps 1 and 2 re: modifying YAML file

File: _dashboards/discover/multi-data-sources.md [hunk 2]
```diff
@@ -132,6 +132,34 @@ The following image shows a date histogram with an interval dropdown list.
...
+    
+    ```
+    # Set the value of this setting to true to enable the experimental multiple data source
+    # support feature. Use with caution.
+    #data_source.enabled: true
+    ```
+
+3. Verify access to OpenSearch Dashboards by connecting to http://localhost:5601 from a browser. The default username and password are admin.
+4. Select **Dev Tools** in the main menu.
+5. Input the following query in the editor pane of the **Console** and then select the play button.
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:150]: "Enter the following query..."

File: _dashboards/discover/multi-data-sources.md [hunk 3]
```diff
@@ -132,6 +132,34 @@ The following image shows a date histogram with an interval dropdown list.
...
+
+    ```json
+    GET /_cat/indices
+    ```
+
+6. From the **DataSource** dropdown menu, select a data source and then query the source.
+7. Repeat the preceding steps for each data source you want to select.
+
+For an overview of this capability, watch this 15-second video.
```
Comment (cwillum) [inline _dashboards/discover/multi-data-sources.md:159]: "For an overview of this process, ..."

**Review by cwillum — APPROVED** [review]
Looks good. Just suggestions.

File: _dashboards/discover/multi-data-sources.md [hunk 4]
```diff
@@ -132,6 +132,34 @@ The following image shows a date histogram with an interval dropdown list.
...
+
+Selecting multiple data sources in the Dev Tools console allows you to work with a broader range of data and gain deeper insights into your code and applications. You can follow these steps:
+
+1. Open `opensearch_dashboards.yml` in the editor of your choice. 
+2. Modify the value `data_source.enabled:` from `false` to `true`, as shown in the following example.
+    
+    ```
+    # Set the value of this setting to true to enable the experimental multiple data source
+    # support feature. Use with caution.
+    #data_source.enabled: true
```
Comment (ariamarble) [inline _dashboards/discover/multi-data-sources.md:145]:
```suggestion
    data_source.enabled: true
```

**Review by ariamarble — APPROVED** [review]
One small change, otherwise looks good to me after Chris's comments

File: _dashboards/dev-tools/index-dev.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+layout: default
+title: Dev Tools
+nav_order: 100
+---
+
+# Dev Tools
+
+**Dev Tools** allows you to set up your OpenSearch Dashboards environment, identify and fix bugs, and customize your Dashboards' appearance and behavior. 
+
+To access the Dev Tools console, select **Dev Tools** in the navigation menu on the OpenSearch Dashboards home page. You'll see an interface like the one shown below.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/dev-tools/index-dev.md:11]: Per UX, the term we should use is menu. I'll look at the example doc too, as that was written before our team.

File: _dashboards/discover/multi-data-sources.md [hunk 6]
```diff
@@ -132,6 +132,33 @@ The following image shows a date histogram with an interval dropdown list.
...
+2. Set `data_source.enabled:` to `true`, as shown in the following example.
+    
+    ```
+    # Set the value of this setting to true to enable the experimental multiple data source
+    # support feature. Use with caution.
+    data_source.enabled: true
+    ```
+
+3. Connect to OpenSearch Dashboards, and select **Dev Tools** in the menu.
+5. Enter the following query in the editor pane of the **Console** and then select the play button.
```
Comment (natebower) [inline _dashboards/discover/multi-data-sources.md:149]:
```suggestion
5. Enter the following query in the editor pane of the **Console** and then select the play button:
```

File: _dashboards/discover/multi-data-sources.md [hunk 7]
```diff
@@ -2,26 +2,26 @@
...
+nav_order: 20
 ---
 
 # Adding multiple data sources
 
 The multiple data sources feature is an experimental feature released in OpenSearch 2.4. It can't be used in a production environment. For updates on the feature’s progress or to leave feedback on improving the feature, see the [OpenSearch Forum discussion](https://forum.opensearch.org/t/feedback-experimental-feature-connect-to-external-data-sources/11144).
 {: .warning }
 
-You can add multiple data sources to a single dashboard. OpenSearch Dashboards allows you to dynamically manage data sources, create index patterns based on those data sources, and execute queries against a specific data source and then combine visualizations in one dashboard.
+ OpenSearch Dashboards allows you to dynamically manage data sources, create index patterns based on those data sources, execute queries against a specific data source, and combine visualizations in one dashboard.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards/discover/multi-data-sources.md:13]: Change made.

**Pull request merged**